src/peft/mapping.py

```diff
@@ -181,6 +182,21 @@ def get_peft_model(
     new_name = model.__dict__.get("name_or_path", None)
     peft_config.base_model_name_or_path = new_name
 
+    # Especially in notebook environments there could be a case that a user
+    # wants to experiment with different configuration values. However, it
+    # is likely that there won't be any changes for new configs on an already
+    # initialized PEFT model. The best we can do is warn the user about it.
+    try:
+        if len(get_layer_status(model)) > 0:
```

Comment:
Nice idea to use `get_layer_status`. I wonder if in this case, a simple

`if any(isinstance(module, BaseTunerLayer) for module in model.modules())`

would not serve the purpose better. This check would stop once the first PEFT layer is found, while `get_layer_status` would do a bunch of more work unnecessarily.

Reply:
For some reason I thought that p-tuning/prompt-tuning are also layer tuners (which they aren't) so I thought it was worthwhile to use the more complex `get_layer_status`. But you're correct, a simple check suffices.